standards/application/segmentation.md

```diff
@@ -0,0 +1,182 @@
+
+## Motivation
+
+Waku Relay deployments typically propagate envelopes up to **150 KB** as per [64/WAKU2-NETWORK - Message](https://rfc.vac.dev/waku/standards/core/64/network#message-size). To support larger application payloads, a segmentation layer is required. This specification enables larger messages by partitioning them into multiple envelopes and reconstructing them at the receiver. Erasure-coded parity segments provide resilience against partial loss or reordering.
+
+## Terminology
+
+- **original message**: the full application payload before segmentation.  
+- **data segment**: one of the partitioned chunks of the original message payload.  
+- **parity segment**: an erasure-coded segment derived from the set of data segments.  
+- **segment message**: a Waku envelope whose `payload` field carries a serialized `SegmentMessageProto`.  
+- **`segmentSize`**: configured maximum size in bytes of each data segment’s `payload` chunk (before protobuf serialization).  
+- **sender public key**: the origin identifier used for indexing persistence.
+
+The key words **“MUST”**, **“MUST NOT”**, **“REQUIRED”**, **“SHALL”**, **“SHALL NOT”**, **“SHOULD”**, **“SHOULD NOT”**, **“RECOMMENDED”**, **“NOT RECOMMENDED”**, **“MAY”**, and **“OPTIONAL”** in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Segmentation
+
+### Sending
+
+When the original payload exceeds `segmentSize`, the sender **MUST**:
+
+- Compute a 32-byte `entire_message_hash = Keccak256(original_payload)`.
+- Split the payload into one or more **data segments**, each of size up to `segmentSize` bytes.
+- Optionally generate **parity segments** using Reed–Solomon erasure coding, at the predefined parity rate.  
+  Implementations **MUST NOT** produce more than 256 total segments (data + parity).
+- Encode each segment as a `SegmentMessageProto` with:
+  - The `entire_message_hash`
+  - Either data-segment indices (`segments_count`, `index`) or parity-segment indices (`parity_segments_count`, `parity_segment_index`)
+  - The raw payload data
+- Send all segments as individual Waku envelopes, preserving application-level metadata (e.g., content topic).
+
+Messages smaller than or equal to `segmentSize` **SHALL** be transmitted unmodified.
+
+### Receiving
+
+Upon receiving a segmented message, the receiver **MUST**:
+
+- Validate each segment according to [Wire Format → Validation](#validation).
+- Persist received segments, indexed by `entire_message_hash` and sender.
```

Comment:
Would "buffer" or "cache" be a better word than "persist", which seems to imply some persistence medium